File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/GUI/FabricaWindows.php
<?php

namespace GUI;

use Componentes\Boton;
use Componentes\Checkbox;
use Componentes\Windows\BotonWindows;
use Componentes\Windows\CheckboxWindows;

class FabricaWindows implements FabricaGUI {
	function crearBoton(): Boton {
		return new BotonWindows;
	}

	function crearCheckbox(): Checkbox {
		return new CheckboxWindows;
	}
}

File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/GUI/FabricaWindows.java
import Componentes.Botones.Boton;
import Componentes.Botones.BotonWindows;
import Componentes.Checkboxes.Checkbox;
import Componentes.Checkboxes.CheckboxWindows;

class FabricaWindows implements FabricaGUI {
	public Boton crearBoton() {
		return new BotonWindows();
	}

	public Checkbox crearCheckbox() {
		return new CheckboxWindows();
	}
}
